### ISSUE TITLE
Provide interface or abstract implementation for other message Broker

### ISSUE THREAD

**Issue by chenyan71** [networknt/light-tram-4j#5]
Currently light-tram-4j use Kafka as message Broker for event/message store. System should Provide interface or abstract implementation for other type of  message Broker
for example: MQ, Chronicle Queue subsystem, Solace...

**Comment by archenroot** [comment]
Hi guys, any progress on this? I am looking to use MQ for broker and MongoDB for event store, are there any abstract interfaces available to implement custom components?
thx.

**Comment by archenroot** [comment]
Regarding the MongoDB CDC compnent I have found following promising project:
https://github.com/networknt/light-tram-4j/issues/5

which is able to tail db log for not only mysql and postgres, but has capability to tail mongodb OPLOG, so good candidate for it.

**Comment by stevehu** [comment]
@archenroot We have used that library before but removed it to use mysql binlog directly as the library is not stable and lack of activity in the GitHub repo. What we are planning is the create an abstract layer for the database and message broker so that users can plug in whatever they are using. It is already on the roadmap but hasn't started yet.

**Comment by archenroot** [comment]
I did some mistake by inserting link with the CDC project:
https://github.com/debezium/debezium

I also asked Richardson at his eventuate about his progress on this. He doesn't like debezium:
https://github.com/eventuate-local/eventuate-local/issues/45

@stevehu - as my link was broken, just to confirm have you refered in your answer to debezium?

@chenyan71 - thx for hint I will take a look.

**Comment by stevehu** [comment]
Yes. I refer the exact library. It was too complex than we need. It looks like the development is picking up recently and I am open to using it if we need to support Postgres and Mongo.
